examples/Demo/Shared/Shared/DemoSearch.razor

```diff
@@ -0,0 +1,82 @@
+﻿@using System.Diagnostics
+@inject DemoNavProvider NavProvider
+@inject NavigationManager NavigationManager
+
+<FluentSearch @bind-Value="@_searchValue"
+              Immediate
+              @bind-Value:after=HandleSearchInput
+              Id="demo-search"
+              Placeholder="Search everything..."/>
+
+@if (_searchResults is not null)
+{
+    <FluentMenu Anchor="demo-search"
+                HorizontalPosition="HorizontalPosition.Right"
+                @bind-Open="_visible">
+        @foreach (var searchResult in _searchResults)
+        {
+            var item = NavProvider.FlattenedMenuItems.First(x => x.Title == searchResult);
+
+            <FluentMenuItem OnClick="@(() => HandleSearchClicked(item))">
+                <span slot="start">
+                    <FluentIcon Value="@(item.Icon)" Class="search-result-icon" Color="Color.Neutral" Slot="start"/>
+                </span>
+                @item.Title
+            </FluentMenuItem>
+        }
+    </FluentMenu>
+}
+
+@code {
```

Comment:
If it is possible, could you separate the razor and the code parts (in a .cs file)? Thanks a lot :-)